framework/src/meshmodifiers/SideSetsAroundSubdomain.C

```diff
@@ -58,6 +62,9 @@ SideSetsAroundSubdomain::modify()
   if (!_mesh_ptr)
     mooseError("_mesh_ptr must be initialized before calling SideSetsAroundSubdomain::modify()!");
 
+  // Initialize the BlockRestrictable parent
+  initializeBlockRestrictable(_pars);
```

Comment:
OK - then I'll go with my solution above. It seems like the BASE CLASS should be doing the following:

```
initialization of the boundary restrictable class if necessary
modify()
calling mesh->prepare() if force
```